ui/dashboards/src/components/Variables/Variable.tsx

```diff
@@ -37,42 +37,88 @@ export function TemplateVariable({ name }: TemplateVariableProps) {
 function ListVariable({ name }: TemplateVariableProps) {
   const ctx = useTemplateVariable(name);
   const definition = ctx.definition as ListVariableDefinition;
-  const { setVariableValue, loadTemplateVariable } = useTemplateVariableActions();
+  const variablePlugin = useVariablePlugin(definition);
+
+  const { setVariableValue, setVariableLoading, setVariableOptions } = useTemplateVariableActions();
   const allowMultiple = definition?.spec.allowMultiple === true;
+  const allowAllValue = definition?.spec.allowAllValue === true;
+
+  const loadOptions = useCallback(async () => {
+    if (!variablePlugin) {
+      return;
+    }
+    setVariableLoading(name, true);
+    try {
+      const { data } = await variablePlugin.getVariableOptions(definition);
+      setVariableOptions(name, data);
+    } catch (e) {
+      console.error('Failed to load template variable options', e);
+    }
+    setVariableLoading(name, false);
+  }, [variablePlugin, definition, name, setVariableLoading, setVariableOptions]);
+
   useEffect(() => {
-    loadTemplateVariable(name);
-  }, [name, loadTemplateVariable]);
+    loadOptions();
+  }, [loadOptions]);
 
   let value = ctx.state?.value;
+  const options = ctx.state?.options;
+  const loading = ctx.state?.loading;
 
+  // Make sure value is an array if allowMultiple is true
   if (allowMultiple && !Array.isArray(value)) {
-    value = [];
+    value = typeof value === 'string' ? [value] : [];
   }
 
+  const finalOptions = useMemo(() => {
+    let computedOptions = options ? [...options] : [];
+
+    // Add the all value if it's allowed
+    if (allowAllValue) {
+      computedOptions = [{ value: DEFAULT_ALL_VALUE, label: 'All' }, ...computedOptions];
+    }
+    return computedOptions;
+  }, [options, allowAllValue]);
+
+  // If there is no value and there are options loaded, select the first value
+  useEffect(() => {
+    const firstOption = finalOptions?.[0];
+    if ((value === null || value?.length === 0) && firstOption) {
+      setVariableValue(name, firstOption.value);
+    }
+  }, [finalOptions, setVariableValue, value, name]);
+
   return (
     <Box display={'flex'}>
       <FormControl>
         <InputLabel id={name}>{name}</InputLabel>
         <Select
+          sx={{ minWidth: 100 }}
           id={name}
           label={name}
-          value={value}
+          value={value ?? ''}
           onChange={(e) => {
-            setVariableValue(name, e.target.value as string);
+            // Must be selected
+            if (e.target.value === null || e.target.value.length === 0) {
+              return;
+            }
+            setVariableValue(name, e.target.value as VariableValue);
           }}
           multiple={allowMultiple}
         >
-          {ctx.state?.options?.map((option) => (
+          {loading && (
```

Comment:
Does this `Select` need to become an `Autocomplete` eventually to support multiple selections? Was thinking about it while looking at loading state because I think that control also has a built-in loading state/indicator.

Reply:
Yeah I think I might need to change this to Autocomplete. I don't think the Select allows you to "create options" on the fly, which is probably needed too. I'll put out a follow up PR if thats the case.